### PR TITLE
Mri upload auto launcher config option

### PR DESCRIPTION
**Pull request by mohadesz**
This pull request includes a new config option called "ImagingUploaderAutoLaunch" which indicates whether the imaging-upload pre-processing scripts should be ran automatically by the ServerProcessLauncher on the front-end or not.

### CODE REVIEW

**Comment by rathisekaran** [comment]
The config option should be added to default schema as well.

File: SQL/0000-00-00-schema.sql [hunk 1]
```diff
@@ -2145,6 +2145,8 @@ INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType,
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'DoubleDataEntryInstruments', "Instruments for which double data entry should be enabled", 1, 1, 'instrument', ID, 'Double data entry instruments', 16 FROM ConfigSettings WHERE Name="study";
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'InstrumentResetting', 'Allows resetting of instrument data', 1, 0, 'boolean', ID, 'Instrument Resetting', 17 FROM ConfigSettings WHERE Name="study";
 INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'SupplementalSessionStatus', 'Display supplemental session status information on Timepoint List page', 1, 0, 'boolean', ID, 'Use Supplemental Session Status', 18 FROM ConfigSettings WHERE Name="study";
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'ImagingUploaderAutoLaunch', "Allows running the ImagingUpload pre-processing scripts", 1, 0, 'boolean', ID, 'ImagingUploader Auto Launch',19 FROM ConfigSettings WHERE Name="Study";
```
Comment (Jkat) [inline SQL/0000-00-00-schema.sql]: Should be 'study' not 'Study'

File: SQL/2015-04-14_imaging_upload_auto_launch_config_option.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+INSERT INTO ConfigSettings (Name, Description, Visible, AllowMultiple, DataType, Parent, Label, OrderNumber) SELECT 'ImagingUploaderAutoLaunch',"Allows running the ImagingUpload pre-processing scripts", 1, 0, 'boolean', ID, 'ImagingUploader Auto Launch', 19 FROM ConfigSettings WHERE Name="Study";
```
Comment (Jkat) [inline SQL/2015-04-14_imaging_upload_auto_launch_config_option.sql]: Should be 'study' not 'Study'

File: modules/mri_upload/php/NDB_Menu_Filter_mri_upload.class.inc [hunk 3]
```diff
@@ -314,6 +314,10 @@ class NDB_Menu_Filter_Mri_Upload extends NDB_Menu_Filter_Form
         $feedbacks = array();
         $db        = Database::singleton();
         $file_name =  $file->fileInfo['name'];
+        $config    = NDB_Config::singleton();
+        $ImagingUploaderAutoLaunch = $config->getSetting(
+            'ImagingUploaderAutoLaunch'
+        );
 
         ///////////////////////////////////////////////////////////////////////
         ///////////////If empty get it using User class////////////////////////
```
Comment (driusan) [inline modules/mri_upload/php/NDB_Menu_Filter_mri_upload.class.inc]: There are a lot of slashes here.

**Comment by rathisekaran** [comment]
Needs to be rebased. Also I tested this pull request, even though the config option was set to 0 the code was trying to insert into server processor table.

**Pull request merged**